### DISCUSSION TITLE
Duck Chess compatible executable

### DISCUSSION THREAD

**Discussion by EdoMachi** [fairy-stockfish/Fairy-Stockfish#581] [Q&A] [answered]
Where can I find and executable that supports Duck Chess?
I thought downloading the artifacts here would do the trick: github.com/ianfab/Fairy-Stockfish/actions/runs/4000320851
But neither 'setoption name UCI_Variant value duck' nor 'xboard', 'variant duck' seem to result in duck chess moves.

**Comment by ianfab** [comment]
So far the automated development builds do not include duck chess support. Duck chess as well as a few other variants with large branching factor (like game of the amazons) are only included when enabling this feature specifically during compilation, see https://github.com/ianfab/Fairy-Stockfish/wiki/Compiling-Fairy-Stockfish#compiling-with-support-for-variants-with-large-branching-factor-duck-amazons, and this is disabled by default since enabling it increases memory consumption and reduces performances for all variants.

I already thought about this but haven't decided on it yet. Neither enabling it by default nor doubling the number of binaries is a good option to me (for both dev builds and releases), so I might maybe include just one single extra binary including these variants instead of providing optimized ones for different hardwares to at least provide some pre-built binaries without cluttering the releases too much.

**Reply by RickGroszkiewicz** [reply]
I am trying to use FSF-Duck for a linux chess match across my network. I can't get HGM's connect program to work properly when I specify the duck chess NNUE file.

I tried to compile FSF-Duck including the NNUE file. Unfortunately, the file that gets included by default is nn-3475407dc199.nnue, which is NOT for Duck chess. 

I tried editing evaluate.h, but I either get error 1 or error 3 near line 826 of the makefile.

**Reply by ianfab** [reply]
If you want to build in a custom NNUE net, you can e.g. follow the steps that the automated NNUE releases for xiangqi/janggi/makruk follow: https://github.com/fairy-stockfish/Fairy-Stockfish-NNUE/blob/60c0aeffef3993ef366a545377545b1887449340/.github/workflows/release.yml#L41-L51

What you might be missing are the `nnue=yes load_net=` arguments in the make command, which ensure that the NNUE is built in, and that it does not try to download the net from the fishtest server.

**Reply by RickGroszkiewicz** [reply]
OOPS - I thought that solved my problem, but it looks like it is using nn-3475407dc199.nnue. My newly compiled executable lost four games in a row.

**Reply by RickGroszkiewicz** [reply]
I deleted all the nnue files in /src, and reverted my changes to evaluate.h. Then I copied the duck*.nnue from /FSFDuck and ran this
sudo make -j build ARCH=x86-64 nnue=yes load_net=duck-ba21f91f5d81.nnue 

but that failed:

/tmp/ccLfgj2d.s: Assembler messages:
/tmp/ccLfgj2d.s:9: Error: file not found: nn-3475407dc199.nnue
lto-wrapper: fatal error: g++ returned 1 exit status
compilation terminated.

**Comment by gbtami** [comment, marked as answer]
Another option is to redirect users to https://github.com/gbtami/Fairy-Stockfish/releases
Pychess fairyfishnet workers use these executables, and they contains all the variants
